r/tests/testthat/test-RecordBatch.R

```diff
@@ -121,8 +121,27 @@ test_that("RecordBatch cast (ARROW-3741)", {
   expect_equal(batch2$column(1L)$type, int64())
 })
 
+test_that("record_batch() handles schema= argument", {
+  s <- schema(x = int32(), y = int32())
+  batch <- record_batch(x = 1:10, y = 1:10, schema = s)
+  expect_equal(s, batch$schema)
+
+  s <- schema(x = int32(), y = float64())
+  batch <- record_batch(x = 1:10, y = 1:10, schema = s)
+  expect_equal(s, batch$schema)
+
+  s <- schema(x = int32(), y = utf8())
+  expect_error(record_batch(x = 1:10, y = 1:10, schema = s))
```

Comment:
If `schema` were a column name I guess you would have to pass the arguments differently